Documentation for -XX:[+|-]HandleSIGUSR2 (#1080)
eclipse-openj9/openj9-docs: docs/openj9_signals.md

- The use of `SIGRTMIN` is configurable with the `-Xdump:suspendwith=<num>` option.
- The handling of `SIGABRT` is configurable with the `-XX[+|-]HandleSIGABRT` option.

> - The handling of `SIGUSR2` is configurable with the [`-XX[+|-]HandleSIGUSR2`](xxhandlesigusr2.md) option.

Reviewer: The colon is missing after `-XX` for both `ABRT` and `USR2`, here and several other places.

Reviewer 2: When fixing, links could also be added for HandleSIGABRT and -Xdump:suspendwith

Reviewer: I'll update https://github.com/eclipse-openj9/openj9-docs/pull/1084.

| Windows                | `jsig.dll`     | Link the DLL with the application that creates or embeds a VM |
| AIX                    | `libjsig.so`   | `cc_r [-q64] <other_compile/link_parameter> -L<java_install_dir> -ljsig -L<java_install_dir>/lib/j9vm -ljvm <java_application>.c` |

> In the absence of signal chaining, VM does not allow application signal handlers for certain signals that are used internally by the VM, including the `SIGUSR2` signal. You can use the `-XX:-HandleSIGUSR2` option instead, whereby the VM signal handler is not installed on VM startup. Therefore, the application signal handler, if available, takes over the handling of the `SIGUSR2` signal. If there is no application signal handler, then the operating system's default signal handler is used.

Author: Done

| Windows                | `jsig.dll`     | Link the DLL with the application that creates or embeds a VM |
| AIX                    | `libjsig.so`   | `cc_r [-q64] <other_compile/link_parameter> -L<java_install_dir> -ljsig -L<java_install_dir>/lib/j9vm -ljvm <java_application>.c` |

> With the `-XX:-HandleSIGUSR2` option, the VM signal handler is not installed even on receiving the `SIGUSR2` signal, which targets the VM. Therefore, even if the signal handler chaining library is not loaded, VM handler does not replace the application signal handler, if available, when handling the `SIGUSR2` signal.

Reviewer 2: VM handler -> the VM handler

Author: Done

> With the `-XX:-HandleSIGUSR2` option, the VM signal handler is not installed even on receiving the `SIGUSR2` signal, which targets the VM. Therefore, even if the signal handler chaining library is not loaded, the VM handler does not replace the application signal handler, if available, when handling the `SIGUSR2` signal.

Reviewer 3: The signal handler is installed on VM startup. It is not installed upon receiving the `SIGUSR2` signal. 
```suggestion
With the `-XX:-HandleSIGUSR2` option, the VM signal handler is not installed. In the absence of the signal handler chaining library, the application signal handler, if available, is not overridden by the VM signal handler for handling the `SIGUSR2` signal.
```
